### DISCUSSION TITLE
I want to use pouchdb at app . how do it ?

### DISCUSSION THREAD

**Discussion by phpsmarter** [neutralinojs/neutralinojs#812] [Q&A]
I want to use pouchdb at app . how do it ?
Whether need adaptor? or something.  seem's it does't work  at app but work fine add create-react-app

**Comment by phpsmarter** [comment]
At now  problem seems solved . 
Directly  add script at index.html
`<script  src="//cdn.jsdelivr.net/npm/pouchdb@7.2.1/dist/pouchdb.min.js"></script>`

```js
//main.js
var localDB = new PouchDB('salttiger2');
var remoteDB = new PouchDB('http://admin:admin@192.168.1.6:5984/salttiger2');
const initDB=async (localDB,remoteDB)=>{
    localDB.sync(remoteDB).on('complete', function () {
          console.log("sync done!")
      }).on('error', function (err) {
         console.log("error")
      });
   
}
```
![](https://tva1.sinaimg.cn/mw690/e6c9d24egy1gzq35lgg3wj20ge07kjro.jpg)
Now I add preact, tailwind , pouchdb . It works. But I still don't get system file system .
Next step. I want to use Pouchdb at React-App?  how to   Global inject db object?

**Reply by pegvin** [reply]
global inject?

**Reply by phpsmarter** [reply]
It it global inject . But I don't know how to inject to create-react-app . maybe add to window object

**Reply by pegvin** [reply]
doesn't `window.dbObject` do the work?

**Reply by phpsmarter** [reply]
Seems now  it works in create-react-app  and using typescript template . 
<database.ts>
```javascript
import PouchDB from 'pouchdb'

const localDB = new PouchDB('salttiger2');
const remoteDB = new PouchDB('http://admin:admin@192.168.1.2:5984/salttiger2');

const initDB=async ()=>{
  localDB.sync(remoteDB).on('complete', function () {
        console.log("sync done!")
    }).on('error', function (err) {
       console.log("error")
    });
 
}

async function  getDbInfo(localDB:any){
  // db.info().then(function (info) {
  //     console.log(info);
  //   })
  let db=localDB;
  try {
      var result = await db.allDocs({
        include_docs: true,
        attachments: true,
        limit:10,
      });
    } catch (err) {
      console.log(err);
    }
  result.rows.map((item:any,index:number)=>{
        console.log(item.doc.title);
  })

  return result.total_rows;
}
```
It is works, don't need  global injection  .  I tried   vanilla.js , preact.js  react.js  . all works now !

**Comment by phpsmarter** [comment]
As for window.dbObject . 
Because of  I want to use  typescript. but window object  have not  pouchdb property.   Now I find solution to give  "any" type to db object.
